turbopack/crates/turbopack-node/js/src/ipc/index.ts

```diff
@@ -96,12 +121,11 @@ function createIpc<TIncoming, TOutgoing>(
   });
 
   function send(message: any): Promise<void> {
-    // Reserve 4 bytes for our length prefix, we will over-write after encoding.
-    const packet = Buffer.from("0000" + JSON.stringify(message), "utf8");
-    packet.writeUInt32BE(packet.length - 4, 0);
-
     return new Promise((resolve, reject) => {
-      socket.write(packet, (err) => {
+      const stringified = JSON.stringify(message)
+      const packet = Buffer.from("0000" + stringified, "utf8");
+      packet.writeUInt32BE(packet.length - 4, 0);
+      socketWritable.write(packet, (err) => {
         process.stderr.write(`TURBOPACK_OUTPUT_D\n`);
```

Comment:
do you understand how these writes are used?

Reply:
sounds good, will do

Reply:
ok, played with this. This is part of figuring out the 'boundaries of a task' and associating console outputs with the loader request.  So these are required otherwise everything times out!.